app/clients/ethereum/ethereum.go

```diff
@@ -46,6 +47,13 @@ func (ec *EthereumClient) WaitForTransactionSuccess(hash common.Hash) (isSuccess
 func (ec *EthereumClient) waitForTransactionReceipt(hash common.Hash) (txReceipt *types.Receipt, err error) {
 	for {
 		_, isPending, err := ec.Client.TransactionByHash(context.Background(), hash)
+
+		// try again mechanism in case transaction is not validated for tx mempool yet
+		if errors.Is(ethereum.NotFound, err) {
```

Comment:
Just an idea -> wouldn't it be easier if we just switch the places for time.Sleep and `TransactionByHash`?

.. doing some work...
.. wait for tx to be mined.. (no need to immediately check for that TX, we can wait 5 sec and then do `TransactionByHash`)